quic/s2n-quic-core/src/transport/error.rs

```diff
@@ -165,6 +166,72 @@ macro_rules! impl_errors {
             }
         }
 
+        impl aggregate::AsVariant for Code {
+            const VARIANTS: &'static [aggregate::info::Variant] = &{
+                use aggregate::info::{Variant, Str};
+
+                const fn count(_v: u64) -> usize {
+                    1
+                }
+
+                const QUIC_VARIANTS: usize = 0 $( + count($code))*;
+
+                const TLS: &'static [Variant] = tls::Error::VARIANTS;
+
+                let mut array = [
+                    Variant { name: Str::new("\0"), id: 0 };
+                    QUIC_VARIANTS + TLS.len() + 1
+                ];
+
+                let mut idx = 0;
+
+                $(
+                    array[idx] = Variant {
+                        name: Str::new(concat!("QUIC_", stringify!($name), "\0")),
+                        id: $code,
+                    };
+                    idx += 1;
+                )*
+
+                let mut tls_idx = 0;
+                while tls_idx < TLS.len() {
+                    let variant = TLS[tls_idx];
+                    array[idx] = Variant {
+                        name: variant.name,
+                        id: variant.id | 0x100,
```

Comment:
maybe assert!(QUIC_VARIANTS < 0x100) or something like that?

Reply:
yeah that makes sense